src/bin-utils/get-script-by-prefix.test.js

```diff
@@ -0,0 +1,224 @@
+  const config = {
+    scripts: {
+      watch: {
+        js: {
+          foo: 'echo foo',
+        },
+      },
+    },
+  }
+  const actual = getScriptByPrefix(config, 'w.j.f')
+  const expected = {
+    script: 'echo foo',
+    name: 'watch.js.foo',
+    description: '',
+  }
+  expect(actual).toEqual(expected)
+})
+
+test(
+  'getScriptByPrefix resolves single prefixes with object values && script key',
+  () => {
+    const config = {
+      scripts: {
+        foo: {
+          description: 'Foo script',
+          script: 'echo foo',
+        },
+      },
+    }
+    const actual = getScriptByPrefix(config, 'f')
+    const expected = {
+      script: 'echo foo',
+      name: 'foo',
+      description: 'Foo script',
+    }
+    expect(actual).toEqual(expected)
+  },
+)
+
+test(
+  'getScriptByPrefix resolves nested prefixes with object values && script key',
+  () => {
+    const config = {
+      scripts: {
+        watch: {
+          foo: {
+            description: 'Foo script',
+            script: 'echo foo',
+          },
+        },
+      },
+    }
+    const actual = getScriptByPrefix(config, 'w.f')
+    const expected = {
+      script: 'echo foo',
+      name: 'watch.foo',
+      description: 'Foo script',
+    }
+    expect(actual).toEqual(expected)
+  },
+)
+
+test('getScriptByPrefix resolves ambiguities with single prefix', () => {
+  const config = {
+    scripts: {
+      foo: {
+        js: {
+          default: 'foo.js script',
+          script: 'echo foo.js',
+        },
+      },
+      foobar: {
+        description: 'Foobar script',
+        script: 'echo foobar',
+      },
+    },
+  }
+  const actual = getScriptByPrefix(config, 'f')
+  const expected = {
+    script: 'echo foobar',
+    name: 'foobar',
+    description: 'Foobar script',
+  }
+  expect(actual).toEqual(expected)
+})
+
+describe('getScriptByPrefix resolves ambiguities with nested prefix', () => {
+  let config
+  beforeEach(() => {
+    config = {
+      scripts: {
+        foo: {
+          bar: {
+            default: 'echo foo.bar',
+            baz: {
+              default: 'echo foo.bar.baz',
+              joe: {
+                default: {
+                  script: 'echo foo.bar.baz.joe',
+                },
+              },
+              invalid: 42,
+              s: {
+                t: 'echo t',
+              },
+            },
+          },
+        },
+        foobar: {
+          description: 'Foobar script',
+          script: 'echo foobar',
+        },
+      },
+    }
+  })
+  test('Foobar works correctly', () => {
+    const actual = getScriptByPrefix(config, 'f')
+    const expected = {
+      script: 'echo foobar',
+      name: 'foobar',
+      description: 'Foobar script',
+    }
+    expect(actual).toEqual(expected)
+  })
+  test('foo.bar works correctly', () => {
+    const actual = getScriptByPrefix(config, 'f.b')
+    const expected = {
+      script: 'echo foo.bar',
+      name: 'foo.bar.default',
```

Comment:
I'm not 100% sure I want to include `.default` in the name... I intended `default` to just be: "If you provide me the prefix long enough to get to this level of nesting and there's no script specified for this level, then execute this one instead."

I'm really not sure. I'm happy to be swayed either direction.